### PR TITLE
Warning fixes

### PR DESCRIPTION
**Pull request by babrodtk**
This fixes some warnings that appear on my setup.

### CODE REVIEW

File: opm/core/grid/GridHelpers.hpp [hunk 1]
```diff
@@ -190,10 +190,6 @@ const double* beginCellVolumes(const UnstructuredGrid& grid);
 /// \brief Get an iterator over the cell volumes of a grid positioned after the last cell.
 const double* endCellVolumes(const UnstructuredGrid& grid);
 
-/// \brief Get the cell centroid of a face.
-/// \param grid The grid whose cell centroid we query.
-/// \param face_index The index of the corresponding face.
-const double* faceCentroid(const UnstructuredGrid& grid, int face_index);
```
Comment (babrodtk) [inline opm/core/grid/GridHelpers.hpp]: This is a duplicate of the FaceCentroidTraits<UnstructuredGrid> version below.

File: opm/core/utility/Units.hpp [hunk 2]
```diff
@@ -216,6 +218,16 @@ namespace Opm
                 return q / unit;
             }
         } // namespace convert
+
+
+#ifndef HAS_ATTRIBUTE_UNUSED
```
Comment (babrodtk) [inline opm/core/utility/Units.hpp]: OPM_UNUSED is a macro defined in opm/material/common/Unused.hpp that perhaps could be refactored into opm-common, and then used here?

Comment (andlaus) [inline opm/core/utility/Units.hpp]: >  that perhaps could be refactored into opm-common, and then used here?

sure. (the reason that it is in opm-material is that it seems like  so far nobody upstream of that module seemed to have an interest in it/was aware of it.)

**Comment by atgeirr** [comment]
Thanks, will merge. I agree we should put Unused.hpp in opm-common, will you do a PR @andlaus? I think opm/common/utility/Unused.hpp is a good path for it.

Since HAS_ATTRIBUTE_UNUSED comes from the config.h though I am not sure it can reliably be used in public headers though?

**Pull request merged**